modules/distributed_topology/boot.tf

```diff
@@ -0,0 +1,109 @@
+############################################################# Boot Nodes ##############################################################
+resource "libvirt_volume" "ubuntu_boot" {
+     name = "ubuntu_boot"
+     pool = "default"
+     source = "${var.boot_img_path}"
+     format = "qcow2"
+}
+
+# Use CloudInit to add our ssh-key to the instance
+resource "libvirt_cloudinit" "boot" {
+       name           = "boot.iso"
+       pool = "default"
+       user_data = "${file("node_config/boot_config")}"
+       local_hostname = "boot0"
+}
+
+# Create the machine
+resource "libvirt_domain" "ICP_boot" {
+     name = "ICP_boot"
+     memory = "5096"
+     vcpu = 2
+     cloudinit = "${libvirt_cloudinit.boot.id}"
+     network_interface {
+           hostname = "boot0"
+           network_name = "default"
+     }
+
+ # IMPORTANT
+ # Ubuntu can hang is a isa-serial is not present at boot time.
+ # If you find your CPU 100% and never is available this is why
+     console {
+           type = "pty"
+           target_port = "0"
+           target_type = "serial"
+     }
+     console {
+             type = "pty"
+             target_type = "virtio"
+             target_port = "1"
+     }
+     disk {
+              volume_id = "${libvirt_volume.ubuntu_boot.id}"
+     }
+     graphics {
+           type = "spice"
+           listen_type = "address"
+           autoport = "true"
+     }
+
+     
+     provisioner "local-exec" {
+           command = "echo 'boot' >> input.txt && echo 'boot0 ${libvirt_domain.ICP_boot.network_interface.0.addresses.0}' >> input.txt "
+     }
+      
+     provisioner "remote-exec" {
+        inline = [
+              "mkdir -p /opt/icp_config",
+            ]
+    }
+
+#    provisioner "remote-exec" {
+ #          script = "./scripts/prereq.sh"
```

Comment:
I believe this should not be comment. We do need prereq.sh to be executed on nodes?